framework/src/media/audio/audio_manager.c

```diff
@@ -1385,6 +1691,10 @@ audio_manager_result_t change_stream_in_device(int card_id, int device_id)
 		return AUDIO_MANAGER_INVALID_PARAM;
 	}
 
+	if ((card_id == g_actual_audio_in_card_id) && device_id == (g_audio_in_cards[g_actual_audio_in_card_id].device_id)) {
```

Comment:
Is this logic only applied to input cards though input and output cases are considered individually in line 1663..?

Reply:
please see change_stream_out_device